src/conn/conn_layered.c

```diff
@@ -505,159 +453,11 @@ __layered_table_get_constituent_cursor(
     return (0);
 }
 
-/*
- * __layered_table_manager_checkpoint_locked --
- *     Trigger a checkpoint of the handle - will acquire necessary locks
- */
-static int
-__layered_table_manager_checkpoint_locked(WT_SESSION_IMPL *session)
-{
-    WT_DECL_RET;
-
-    WT_STAT_CONN_DSRC_INCR(session, layered_table_manager_checkpoints);
-    WT_WITH_CHECKPOINT_LOCK(
-      session, WT_WITH_SCHEMA_LOCK(session, ret = __wt_checkpoint(session, 0)));
-    return (ret);
-}
-
-/*
- * __layered_table_manager_checkpoint_one --
- *     Review the layered tables and checkpoint one if it has enough accumulated content. For now
- *     this just checkpoints the first table that meets the threshold. In the future it should be
- *     more fair in selecting a table.
- */
-static int
-__layered_table_manager_checkpoint_one(WT_SESSION_IMPL *session)
-{
-    WT_BTREE *ingest_btree;
-    WT_CURSOR *stable_cursor;
-    WT_DECL_RET;
-    WT_LAYERED_TABLE_MANAGER *manager;
-    WT_LAYERED_TABLE_MANAGER_ENTRY *entry;
-    WT_TXN_ISOLATION saved_isolation;
-    uint64_t satisfied_txn_id;
-    uint32_t i;
-
-    manager = &S2C(session)->layered_table_manager;
-
-    /* The table count never shrinks, so this is safe. It probably needs the layered table lock */
-    for (i = 0; i < manager->open_layered_table_count; i++) {
```

Comment:
We can get rid of this, but I think we will want to bring something similar back at some stage.
The checkpoint here is really to define the cadence at which content in ingest tables becomes obsolete. That doesn't tie well to how often MongoDB creates checkpoints, but it rather ties to how often this standby gets a new checkpoint in the shared table.
Phrased another way, I imagine that we will want a utility thread to manage some aspect of layered tables, so I think we should keep this structure here for now, even if it doesn't need to do anything at the moment.

Reply:
bq. which content in ingest tables becomes obsolete
I thought we do this through timestamps. Does using the checkpoint timestamp enough for this purpose?